sky_sync/open_weather_api/comm.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+import os
+
+import requests
+from requests import Response
+from requests.exceptions import ConnectionError, RequestException
+from rest_framework.exceptions import ParseError
+
+from .api_urls import (
+    build_current_weather_url,
+    build_direct_geocoding_url,
+    build_forecast_weather_url,
+    build_ip_url,
+    build_reverse_geocoding_url,
+)
+
+
+API_TOKEN = os.environ["OPEN_WEATHER_API_KEY"]
+
+
+def handle_request(url: str) -> Response:
+    """Handles GET requests with possible errors."""
+    try:
+        resp = requests.get(url=url)
+    except (RequestException, ConnectionError):
+        print("[bold red]An error occurred. Please check your network connection and try again.[/]")
+        exit(1)
```

Comment:
nie używaj bezpośrednio exit() w kodzie, niech wyjdzie w kodzie będzie naturalną wynikową operacji, które przeprowadzisz, np. return

Reply:
Będę chciał to omówić we wtorek, więc na razie w tym nie grzebię.